x-pack/solutions/observability/plugins/slo/server/services/get_slo_stats_overview.ts

```diff
@@ -8,15 +8,23 @@
 import type { RulesClientApi } from '@kbn/alerting-plugin/server/types';
 import type { IScopedClusterClient } from '@kbn/core-elasticsearch-server';
 import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { type KueryNode, nodeBuilder } from '@kbn/es-query';
 import type { Logger } from '@kbn/logging';
 import { AlertConsumers, SLO_RULE_TYPE_IDS } from '@kbn/rule-data-utils';
 import type { AlertsClient } from '@kbn/rule-registry-plugin/server';
 import type { GetSLOStatsOverviewParams, GetSLOStatsOverviewResponse } from '@kbn/slo-schema';
+import type {
+  AggregationsAggregate,
+  FieldValue,
+  QueryDslQueryContainer,
+} from '@elastic/elasticsearch/lib/api/types';
 import moment from 'moment';
 import { typedSearch } from '../utils/queries';
 import { getSummaryIndices, getSloSettings } from './slo_settings';
 import { getElasticsearchQueryOrThrow, parseStringFilters } from './transform_generators';
 
+const ES_PAGESIZE_LIMIT = 5000;
```

Comment:
Why did we pick 5000 for the bucket limit for the query? If there's no reason we may want to reduce this to 1000, per the [docs](https://www.elastic.co/docs/reference/aggregations/search-aggregations-bucket-composite-aggregation#_pagination) for the composite agg:

> If all composite buckets should be retrieved it is preferable to use a small size (100 or 1000 for instance) and then use the after parameter to retrieve the next results.

Given this won't change any aspect of the implementation, if there's no real reason for the 5k limit we may want to reduce this to 1k.